ansible/library/bgp_route.py

```diff
@@ -0,0 +1,278 @@
+        ## search_state = ['header', 'path_prop', 'prefix_prop1', 'prefix_prop2', 'paths', 'path_l1', 'path_l2', 'path_l2', 'path_l3', 'err' ]
+        state = 'header'
+        self.facts['bgp_rt'][prefix]['paths'] = []
+        while len(result_lines) != 0:
+            line = result_lines.pop(0)
+            #self.facts[prefix]['line'].append(line)
+            if line == '':
+                continue
+            if state == 'header':
+                if regex_prefix_header.match(line) and (prefix not in line):
+                    self.facts['bgp_rt'][prefix]['found'] = False
+                    return
+                if regex_prefix_header.match(line) and prefix in line:
+                    self.facts['bgp_rt'][prefix]['found'] = True
+                    state = 'path_prop'
+                else:
+                    state = 'err'
+            elif state == 'path_prop':
+                find_path = regex_prefix_avail_paths.match(line)
+                if find_path:
+                    path_num = find_path.group(1)
+                    self.facts['bgp_rt'][prefix]['path_num'] = path_num
+                    state = 'path_prefix_prop1'
+                else:
+                    state = 'err'
+            elif  state == 'path_prefix_prop1':
+                state = 'path_prefix_prop2'
+            elif  state == 'path_prefix_prop2':
+                if regex_prefix_prop2.match(line):
+                    state = 'paths'
+                else:
+                    state = 'err'
+            elif state == 'paths':
+                if regex_prefix_paths.match(line):
+                    path = line.strip()
+                    self.facts['bgp_rt'][prefix]['paths'].append(path)
+                    state = 'path_l1'
+                else:
+                    state = 'err'
+            elif state == 'path_l1':
+                if regex_prefix_path_line1.match(line):
+                    state = 'path_l2'
+                else:
+                    state = 'err'
+            elif state == 'path_l2':
+                if regex_prefix_path_line2.match(line):
+                    state = 'path_l3'
+                else:
+                    state = 'err'
+            elif state == 'path_l3':
+                if regex_prefix_path_line3.match(line):
+                    state = 'paths'
+                else:
+                    state = 'err'
+            elif state == 'err':
+                raise Exception("cannot parse bgp prefix info correctly" + str(self.facts))
+
+
+def collect_data(module, command=None):
+   """
+       Collect bgp information by reading output of 'vtysh' command line tool
+   """
+   rc, out, err = module.run_command('vtysh -c "show ip bgp ' + command + '"',
+                                        executable='/bin/bash', use_unsafe_shell=True)
+   return (rc, out, err)
+
+
+def main():
+    module = AnsibleModule(
+            argument_spec=dict(
+               show_cmd=dict(required=False, default=''),
```

Comment:
can you define parameter like
neighbor=?
direction=(adv/rcv)
prefix=?

direction is only available when neighbor is specified.

It it better for you to ask user to give parameter instead of letting use to give you the raw commands.

Reply:
changed with the suggestion